GOVERNANCE_DOC.md

```diff
@@ -0,0 +1,597 @@
+
+### Validator’s punishment for non-voting
+
+Validators are required to vote on all proposals to ensure that results have legitimacy. Voting is part of validators' directives and failure to do it will result in a penalty. 
+
+If a validator’s address is not in the list of addresses that voted on a proposal and if the vote is closed (i.e. `MinDeposit` was reached and `Voting period` is over), then this validator will automatically be partially slashed of `GovernancePenalty`.
+
+*Note: Need to define values for `GovernancePenalty`*
+
+**Exception:** If a proposal is a `Urgent` proposal and is accepted via the special condition of having a ratio of `Yes` votes to `InitTotalVotingPower` that exceeds 2/3, validators cannot be punished for not having voted on it. That is because the proposal will close as soon as the ratio exceeds 2/3, making it mechanically impossible for some validators to vote on it.
+
+### Governance key and governance address
+
+Validators can make use of an additional slot where they can designate a `Governance PubKey`. By default, a validator's `Governance PubKey` will be the same as its main PubKey. Validators can change this `Governance PubKey` by sending a `Change Governance PubKey` transaction signed by their main `Consensus PubKey`. From there, they will be able to sign vote using the `Governance PrivKey` associated with their `Governance PubKey`. The `Governance PubKey` can be changed at any moment.
+
+
+## Software Upgrade
+
+If proposals are of type `SoftwareUpgradeProposal`, then nodes need to upgrade their software to the new version that was voted. This process is divided in two steps.
+
+### Signal
+
+After a `SoftwareUpgradeProposal` is accepted, validators are expected to download and install the new version of the software while continuing to run the previous version. Once a validator has downloaded and installed the upgrade, it will start signaling to the network that it is ready to switch by including the proposal's `proposalID` in its *precommits*.(*Note: Confirmation that we want it in the precommit?*)
+
+Note: There is only one signal slot per *precommit*. If several `SoftwareUpgradeProposals` are accepted in a short timeframe, a pipeline will form and they will be implemented one after the other in the order that they were accepted.
+
+### Switch
+
+Once a block contains more than 2/3rd *precommits* where a common `SoftwareUpgradeProposal` is signaled, all the nodes (including validator nodes, non-validating full nodes and light-nodes) are expected to switch to the new version of the software. 
```

Comment:
I'm curious if it should be a bit higher than 2/3rd.... consider:
 - if it's an incompatible hard fork and exactly 2/3rds are signalling
 - all non signalling validators will be submitting invalid responses 
 - if a single validator goes offline from the upgraded subset, the entire hub will halt

Why don't we do something higher than 2/3 say 3/4

thoughts?

Reply:
Yes makes sense to me. Should be just a bit higher then, not too much. Waiting for other people to weigh in before modifying the spec @jaekwon @adrianbrink @sunnya97

Reply:
Yes, I think at least 3/4 should be ready.
I mean, bitcoin was aiming for 90% for segwit.

Once a vote passes and some validators refuse to upgrade, in an attempt to veto with eg. 26%, people can re-delegate to ensure that the validators who already upgraded have at least 3/4 power... this upgrade process err on safety rather than speed.